### ISSUE TITLE
apikey is null on domain lookup

### ISSUE THREAD

**Issue by tonnyorg** [centralnicgroup-opensource/rtldev-middleware-whmcs#275]
**Describe the bug**
If you configure InternetBS module as the lookup provider in the `/admin/configdomains.php`, the lookup doesn't fetch IBS API because `apikey` parameter is missing, and that is caused because the `ibs_CheckAvailability` function's `$params` variable doesn't include the expected `$params['Username']` and `$params['Password']` in the `ibs_call` function.

**To Reproduce**
1. Go to `/admin/configdomains.php`
2. Under the `Lookup Provider` section, click on 'change'
3. Select `Domain Registrar` and under `Choose a registrar...` section pick `Internet.bs v4.0.30`
4. In the next screen select all the TLDs and save
5. Edit the `ibs.php` file and var_dump:
```php
function ibs_domainCheck($params)
{
    $domainName = ibs_getInputDomain($params);

    $result = ibs_call($params, "Domain/Check", [
        "domain" => $domainName,
        "currency" => "USD"
    ]);
var_dump([
    'params' => $params,
    'result'=> $result,
]);
die();
```

6. Open the chrome's web inspector (so you can see the var_dump in the network tab) and perform any domain lookup in the WHMCS' public page.

**Expected behavior**
It should perform the domain lookup properly and not fallback on default whois

**Screenshots**
![Screenshot 2023-12-19 at 9 02 56 AM](https://github.com/centralnicgroup-opensource/rtldev-middleware-whmcs/assets/4955186/fb01aaa4-995d-4977-a040-9dc9ac25cc9e)
![Screenshot 2023-12-19 at 9 07 09 AM](https://github.com/centralnicgroup-opensource/rtldev-middleware-whmcs/assets/4955186/d3ddd536-93ea-4995-9b82-58cf5f103712)

var_dump output:
```
array(2) { ["params"]=> array(10) { ["whmcsVersion"]=> string(15) "8.7.2-release.1" ["sld"]=> string(8) "19216813" ["tlds"]=> array(1) { [0]=> string(4) ".xyz" } ["searchTerm"]=> string(8) "19216813" ["tldsToInclude"]=> array(1) { [0]=> string(4) ".xyz" } ["isIdnDomain"]=> bool(false) ["punyCodeSearchTerm"]=> string(0) "" ["premiumEnabled"]=> bool(true) ["original"]=> array(7) { ["sld"]=> string(8) "19216813" ["tlds"]=> array(1) { [0]=> string(4) ".xyz" } ["searchTerm"]=> string(8) "19216813" ["tldsToInclude"]=> array(1) { [0]=> string(4) ".xyz" } ["isIdnDomain"]=> bool(false) ["punyCodeSearchTerm"]=> string(0) "" ["premiumEnabled"]=> bool(true) } ["domainname"]=> string(12) "19216813.xyz" } ["result"]=> array(4) { ["transactid"]=> string(32) "a1b1bcaf207216aab00fb4e7d55518b8" ["status"]=> string(7) "FAILURE" ["message"]=> string(36) "Required parameter "apikey" missing!" ["code"]=> string(6) "100003" } }
```

**Context (please complete the following information):**
 - Device: Any
 - Browser Any
 - WHMCS Version 8.7.2
 - Server OS: Docker/Ubuntu
 - PHP Version 8.1
 - Registrar: IBS
 - Module Version: 4.0.30

**Comment by KaiSchwarz-cnic** [comment]
Good Morning @tonnyorg !

I wonder that we don't see anything related to Registrar Module Configuration Settings in your dump of `$params`.
Tested this in our dev container:
<img width="466" alt="image" src="https://github.com/centralnicgroup-opensource/rtldev-middleware-whmcs/assets/229425/ef7fb36c-b898-412c-af78-72db76cd08b6">
<img width="522" alt="image" src="https://github.com/centralnicgroup-opensource/rtldev-middleware-whmcs/assets/229425/67cc4260-48cb-4e95-b857-eb9a2ed89cfe">
<img width="941" alt="image" src="https://github.com/centralnicgroup-opensource/rtldev-middleware-whmcs/assets/229425/d59c8bca-855a-4bbf-8da1-dcfbd98169ff">

As you see, it works perfectly well.

Can you double check to have the Registrar Module configured? Ensure to have our custom internet.bs module configured (v4.0.30 is part of the module label) and not the one shipped with WHMCS.

If all looks good on your end, please share a screenshot of the registrar module configuration with us for reference.
As this includes sensible information, please reach out to internet.bs support via https://internetbs.net/en/contact.html and share the screenshot with us.

**Comment by tonnyorg** [comment]
@KaiSchwarz-cnic thanks for your response, I've double checked so far looks good, idk what could be wrong

I'm running WHMCS Version 8.7.2, are you running any beta or other version?

![Screenshot 2023-12-20 at 8 29 33 AM](https://github.com/centralnicgroup-opensource/rtldev-middleware-whmcs/assets/4955186/4d7ed8f6-5e7a-4703-a732-becd5403bf33)
![Screenshot 2023-12-20 at 8 47 02 AM](https://github.com/centralnicgroup-opensource/rtldev-middleware-whmcs/assets/4955186/a363d480-96f8-4c17-9096-2cc87f9dc201)
![Screenshot 2023-12-20 at 8 47 47 AM](https://github.com/centralnicgroup-opensource/rtldev-middleware-whmcs/assets/4955186/b2e9542f-7ff1-4a55-991f-5ef33630f3b3)

**Comment by KaiSchwarz-cnic** [comment]
The Registrar Module you've configured isn't our Module. Maybe you upgraded WHMCS? It is then resetting the module to the one shipped with WHMCS. Please reinstall our version.

**Comment by tonnyorg** [comment]
I mean... the module says `Internet.bs v4.0.30` in my screen shoot above... it's installed under `modules/registrars/ibs` folder, but I also noticed the one that comes by default it's under `modules/registrars/internetbs`

**Comment by KaiSchwarz-cnic** [comment]
Sorry for replying that late... vacation followed by sick leave.

As mentioned, there must be something wrong with the registrar module installation on your end. The logo of the registrar module shared by you is the logo of the module shipped with whmcs and not our module.

Compare with our one:
![image](https://github.com/centralnicgroup-opensource/rtldev-middleware-whmcs/assets/229425/63f3c36e-35d1-430d-944f-af63dab9bf87)

It feels like a bad mixture of things here, I fully agree. At least check this please again. Also strange that no settings are present in $params. I can't do more than asking you to save the settings again. If that still does not help, I highly recommend reaching out to WHMCS Support. Having no settings in $params is that strange, I've never seen that on my end. I mean, it looks like you have the module configured, so settings must be present. We are not responsible for loading, saving and providing the settings - WHMCS is doing that.

**Comment by KaiSchwarz-cnic** [comment]
I am closing this case. Still, feel free to follow-up to continue the discussion. I can just ensure this isn't an issue of our module especially because others are using it as well. The problem must be specific to your installation as mentioned.

**Comment by KaiSchwarz-cnic** [comment]
Ehm, sorry for the late follow-up. I had a bit too much around. I noticed now this comment of you:

> I mean... the module says Internet.bs v4.0.30 in my screen shoot above... it's installed under modules/registrars/ibs folder, but I also noticed the one that comes by default it's under modules/registrars/internetbs

Yes exactly, you've the wrong one configured (and eventually in use). Ensure to configure the module with the logo I shared. Whois Lookup provider is already correctly configured. Then also ensure that the Domain Pricing Overview's TLD configuration is pointing also to the right module. Domains you manage over internet.bs have also to point to the right module.

When done, all should work fine.
